### PR TITLE
v118, v119, v120

### PR DESCRIPTION
**Pull request by alexmcdermid**

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/graddatacollection/api/rules/demographic/ruleset/V118DemographicStudentStatus.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    public V118DemographicStudentStatus(DemographicRulesService demographicRulesService) {
+        this.demographicRulesService = demographicRulesService;
+    }
+
+    @Override
+    public boolean shouldExecute(StudentRuleData studentRuleData, List<DemographicStudentValidationIssue> validationErrorsMap) {
+        log.debug("In shouldExecute of StudentStatus-V118: for demographicStudentID :: {}", studentRuleData.getDemographicStudentEntity().getDemographicStudentID());
+
+        var shouldExecute =  isValidationDependencyResolved("V18", validationErrorsMap);
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/graddatacollection/api/rules/demographic/ruleset/V118DemographicStudentStatus.java:41]: V118

File: api/src/main/java/ca/bc/gov/educ/graddatacollection/api/rules/demographic/ruleset/V118DemographicStudentStatus.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                !(
+                    demStudent.getStudentStatusCode().equalsIgnoreCase(student.getStatusCode()) ||
+                    ("A".equalsIgnoreCase(student.getStatusCode()) && "T".equalsIgnoreCase(demStudent.getStudentStatusCode()))
+                )
+            ) {
+                log.debug("StudentStatus-V118: Student Status must match PEN.  demographicStudentID :: {}", demStudent.getDemographicStudentID());
+                errors.add(createValidationIssue(StudentValidationIssueSeverityCode.ERROR, DemographicStudentValidationFieldCode.STUDENT_STATUS, DemographicStudentValidationIssueTypeCode.STUDENT_STATUS_PEN_MISMATCH));
+            }
+        } catch (GradDataCollectionAPIRuntimeException e) {
+            log.debug("StudentStatus-V118:Student No student api student record found for demographicStudentID: {}", demStudent.getDemographicStudentID());
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/graddatacollection/api/rules/demographic/ruleset/V118DemographicStudentStatus.java:67]: We'll need to rethrow the exception so the validation fails and retries later

File: api/src/main/java/ca/bc/gov/educ/graddatacollection/api/service/v1/DemographicRulesService.java [hunk 3]
```diff
@@ -13,7 +16,47 @@ public class DemographicRulesService {
 
     private final RestUtils restUtils;
 
-    public GradStudentRecord getGradStudentRecord(UUID studentID){
-        return restUtils.getGradStudentRecordByStudentID(UUID.randomUUID(), studentID);
+    public Student getStudentApiStudent(String pen) {
+        if (pen == null || pen.isBlank()) {
```
Comment (arcshiftsolutions) [inline api/src/main/java/ca/bc/gov/educ/graddatacollection/api/service/v1/DemographicRulesService.java:20]: This is impossible - we reject entire files with no PEN.

File: api/src/main/java/ca/bc/gov/educ/graddatacollection/api/rules/demographic/ruleset/V118DemographicStudentStatus.java (hunk 2 above)
Comment (alexmcdermid) [reply api/src/main/java/ca/bc/gov/educ/graddatacollection/api/rules/demographic/ruleset/V118DemographicStudentStatus.java:67]: remove try catch so original throw goes through

**Review by arcshiftsolutions — APPROVED** [review]

**Pull request merged**